libcontainer/sync_unix.go

```diff
@@ -42,8 +43,21 @@ func (s *syncSocket) WritePacket(b []byte) (int, error) {
 }
 
 func (s *syncSocket) ReadPacket() ([]byte, error) {
-	size, _, err := unix.Recvfrom(int(s.f.Fd()), nil, unix.MSG_TRUNC|unix.MSG_PEEK)
-	if err != nil {
+	var (
+		size int
+		err  error
+	)
+
+	for {
+		size, _, err = unix.Recvfrom(int(s.f.Fd()), nil, unix.MSG_TRUNC|unix.MSG_PEEK)
+		if err == nil {
+			break
+		}
```

Comment:
Isn't it simpler if we just test for err != unix.EINTR --> break? Then we just need to do the recvfrom and this check in the loop.